### PR TITLE
Create SHOULD_RAISE operator for Sim

### PR DESCRIPTION
**Pull request by smcclellan7**

### CODE REVIEW

File: smrt-server-sim/src/main/scala/com/pacbio/simulator/steps/ConditionalSteps.scala [hunk 1]
```diff
@@ -26,10 +28,27 @@ trait ConditionalSteps {
     }
   }
 
+  case class ExpectErrorStep(step: Step) extends Step {
+    override val name = s"ExpectError-${step.name}"
+    override def run: Future[Result] = step.run.recoverWith {
+      case NonFatal(ex) => Future { EXCEPTION(ex) }
```
Comment (mpkocher) [inline smrt-server-sim/src/main/scala/com/pacbio/simulator/steps/ConditionalSteps.scala]: Are there any concerns here with false positives?

Comment (smcclellan7) [inline smrt-server-sim/src/main/scala/com/pacbio/simulator/steps/ConditionalSteps.scala]: Like, do you want to filter on which exceptions are raised?  E.g., SHOULD_RAISE(IllegalArgumentException)

Comment (smcclellan7) [inline smrt-server-sim/src/main/scala/com/pacbio/simulator/steps/ConditionalSteps.scala]: PTAL

**Comment by mpkocher** [comment]
Looks reasonable. It might be better to do this with partial functions versus reflection, but this can be addressed in a future iteration. I want to get this in and unblock testing efforts for the delete job for @natechols.

**Pull request merged**